### PR TITLE
patch nuget dependencies versions

### PR DESCRIPTION
**Pull request by sergii-s**

### CODE REVIEW

File: src/FullBuild/Core/Publishers.fs [hunk 1]
```diff
@@ -11,98 +11,115 @@
...
+    let packageNuspec = 
+        Env.GetFolder Folder.Package
+        |> GetSubDirectory packageName
+        |> GetFile(sprintf "%s.nuspec" packageName)
+        |> fun x -> x.FullName
+        |> Nuget.Load
+    match packageNuspec.Package, packageNuspec.Package2 with
+    | Some pkg, _ -> pkg.Metadata.Version
+    | _, Some pkg -> pkg.Metadata.Version
+    | _ -> failwith "Unnable to parse nuspec"
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: spelling ;-)

File: src/FullBuild/Core/Publishers.fs [hunk 2]
```diff
@@ -11,98 +11,115 @@
...
-
-type private PublishApp =
+type private PublishApp = 
     { Name : string
       App : Application }
 
+type Nuget = FSharp.Data.XmlProvider< "Examples/Library.nuspec", Global=true, SampleIsList=true >
 
-let private publishCopy (app : PublishApp) =
+let private publishCopy (app : PublishApp) =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: no need to reformat code

File: src/FullBuild/Core/Publishers.fs [hunk 3]
```diff
@@ -11,98 +11,115 @@
...
             let artifactDir = appDir |> GetSubDirectory app.Name
             Bindings.UpdateArtifactBindingRedirects artifactDir
-        else
-            printfn "[WARNING] Can't publish application %A without repository" app.Name
+        else printfn "[WARNING] Can't publish application %A without repository" app.Name
 
-let private publishZip (app : PublishApp) =
-    let tmpApp = { Name = ".tmp-" + app.Name
-                   App = app.App }
+let private publishZip (app : PublishApp) =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: no need to reformat code

File: src/FullBuild/Core/Publishers.fs [hunk 4]
```diff
@@ -11,98 +11,115 @@
...
     if targetFile.Exists then targetFile.Delete()
+    System.IO.Compression.ZipFile.CreateFromDirectory
+        (sourceFolder.FullName, targetFile.FullName, Compression.CompressionLevel.Optimal, false)
 
-    System.IO.Compression.ZipFile.CreateFromDirectory(sourceFolder.FullName, targetFile.FullName, Compression.CompressionLevel.Optimal, false)
-
-let private publishDocker (app : PublishApp) =
-    let tmpApp = { Name = ".tmp-docker"
-                   App = app.App }
+let private publishDocker (app : PublishApp) =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: no need to reformat code

File: src/FullBuild/Core/Publishers.fs [hunk 5]
```diff
@@ -11,98 +11,115 @@
...
     if targetFile.Exists then targetFile.Delete()
-
     let dockerArgs = sprintf "build -t %s ." app.Name
     Exec.Exec "docker" dockerArgs sourceFolder Map.empty |> checkErrorCode
     sourceFolder.Delete(true)
 
-let private publishNuget (app : PublishApp) =
-    let tmpApp = { Name = ".tmp-nuget-" + app.Name
-                   App = app.App }
+let private getDependencyVersion packageName =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: Could this be moved to PackageRelationship.fs ?

File: src/FullBuild/Core/Publishers.fs [hunk 6]
```diff
@@ -11,98 +11,115 @@
...
+        |> GetSubDirectory packageName
+        |> GetFile(sprintf "%s.nuspec" packageName)
+        |> fun x -> x.FullName
+        |> Nuget.Load
+    match packageNuspec.Package, packageNuspec.Package2 with
+    | Some pkg, _ -> pkg.Metadata.Version
+    | _, Some pkg -> pkg.Metadata.Version
+    | _ -> failwith "Unnable to parse nuspec"
+
+let private updateDependencies (nuspecFile : FileInfo) =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: Could this be moved to Generators/Packages.fs ?

File: src/FullBuild/Core/Publishers.fs [hunk 7]
```diff
@@ -11,98 +11,115 @@
...
         Exec.Exec "nuget" nugetArgs sourceFolder Map.empty |> Exec.CheckResponseCode
         targetFolder.Create()
-        for file in sourceFolder.EnumerateFiles("*.nupkg") do 
+        for file in sourceFolder.EnumerateFiles("*.nupkg") do
             file.MoveTo(Path.Combine(targetFolder.FullName, file.Name)) |> ignore
         sourceFolder.Delete(true)
     | None -> failwith (sprintf "No nuspec found for the application %s" app.Name)
 
-let PublishWithPublisher (app : Application) =
+let PublishWithPublisher(app : Application) =
```
Comment (pchalamet) [inline src/FullBuild/Core/Publishers.fs]: no need to reformat code

File: src/FullBuild/IOHelpers.fs [hunk 8]
```diff
@@ -149,6 +149,9 @@ let FindKnownProjects (repoDir : DirectoryInfo) =
 let EnumarateFiles (dir : DirectoryInfo) =
     dir.EnumerateFiles()
 
+let EnumarateChilds (dir : DirectoryInfo) =
```
Comment (pchalamet) [inline src/FullBuild/IOHelpers.fs]: EnumerateChildren instead

File: src/FullBuild/IOHelpers.fs [hunk 9]
```diff
@@ -149,6 +149,9 @@ let FindKnownProjects (repoDir : DirectoryInfo) =
 let EnumarateFiles (dir : DirectoryInfo) =
     dir.EnumerateFiles()
 
+let EnumarateChildren (dir : DirectoryInfo) =
```
Comment (pchalamet) [inline src/FullBuild/IOHelpers.fs]: EnumerateChildren

e instead of a ;-)

Comment (sergii-s) [inline src/FullBuild/IOHelpers.fs]: pfff didn't notice

File: src/FullBuild/Generators/Packagers.fs [hunk 10]
```diff
@@ -0,0 +1,24 @@
+﻿module Generators.Packagers
+
+open System.IO
+
+type Nuget = FSharp.Data.XmlProvider< "Examples/Library.nuspec", Global=true, SampleIsList=true >
+
+let private getDependencyVersion package =
```
Comment (pchalamet) [inline src/FullBuild/Generators/Packagers.fs]: There is already a function either in Parsers/Packages.fs or Parsers/MsBuild.fs that does this.

Could that be possible to merge both of them ?

**Comment by pchalamet** [comment]
merged in develop. It's such a pita to work with gitflow.

**Comment by pchalamet** [comment]
release with v4.0

**Pull request closed without merging**